File: JavaScript/BoardView.js


function BoardView(board) {
	var width  = board.getWidth();
	var height = board.getHeight();
		
	this.setBoardData = function(board) {
		this.board = board;
	}
	
	this.updateCompletely = function() {
		for (var x in this.board) {
			for (var y in this.board[x]) {
				var apple = this.board[x][y];
				Terminal.cursorTo(apple.x + 1, apple.y + 1);
				Terminal.put('o');
			}
		}
	}
	
	this.remove = function(element) {
		Terminal.cursorTo(element.x + 1, element.y + 1);
		Terminal.put(' ');
	}
	
	this.add = function(element) {
		Terminal.cursorTo(element.x + 1, element.y + 1);
		Terminal.put('#');
	}
	
	this.addApple = function(apple) {
		Terminal.cursorTo(apple.x + 1, apple.y + 1);
		Terminal.put('o');
	}
	
	function strRepeat(str, times) {
		var result = new String();
		while (times > 0) {
			times--;
			result += str;
		}
		return result;
	}
	
	this.drawBoarder = function() {
		Terminal.cursorTo(0, 0);
		Terminal.put('/' + strRepeat('-', width) + '\\\n\r');
		
		
		for (i = 1; i < height + 1; i++) {
			//Terminal.cursorTo(0, i);
			//Terminal.put('|');
			//Terminal.cursorTo(this.width + 1, i);
			Terminal.put('|' + strRepeat(' ', width) + '|\n\r');
		}
		//Terminal.put("\n");
		Terminal.put('\\' + strRepeat('-', width) + '/');
	}
	
	this.drawBoarder();
	board.setView(this);
}

File: Ruby/board_view.rb
# 
# To change this template, choose Tools | Templates
# and open the template in the editor.
 

class BoardView
  def initialize board
    board.view = self
    @width = board.width
    @height = board.height
  end
  
  def boardData= board
    @board = board
  end
  
  def updateCompletely
    @board.each { |y|
      if y
        y.each { |apple|
          if apple
            addApple apple
          end
        }
      end
    }
  end
  
  def remove(element)
    Terminal.cursorTo(element.x + 1, element.y + 1)
    Terminal.put ' '
  end
  
  def add(element)
    Terminal.cursorTo(element.x + 1, element.y + 1)
    Terminal.put '#'
  end
  
  def addApple(apple)
    Terminal.cursorTo(apple.x + 1, apple.y + 1)
    Terminal.put 'o'
  end
  
  def drawBoarder
    Terminal.cursorTo 0, 0
    Terminal.put '/'
    @width.times { Terminal.put '-' }
    Terminal.put '\\'
    
    for i in 1..@height do
      Terminal.cursorTo 0, i
      Terminal.put '|'
      Terminal.cursorTo @width + 1, i
      Terminal.put '|'
      Terminal.put "\n"
    end
    
    Terminal.put '\\'
    @width.times { Terminal.put '-' }
    Terminal.put '/'
  end
end